test/irb/test_irb.rb

```diff
@@ -825,6 +825,13 @@ def build_irb
   end
 
   class BacktraceFilteringTest < TestIRB::IntegrationTestCase
+    def setup
+      super
+      # These tests are sensitive to warnings, so we disable them
+      original_rubyopt = ENV["RUBYOPT"] || "" + (@envs["RUBYOPT"] || "")
```

Comment:
Precedence of the operator will be `ENV["RUBYOPT"] || ("" + (@envs["RUBYOPT"] || ""))`

How about this? It will make it chainable in subclasses. (expecting `@envs["RUBYOPT"]` already including `ENV["RUBYOPT"]` if it is not nil)
```ruby
original_rubyopt = @envs["RUBYOPT"] || ENV["RUBYOPT"] || ""
```

Reply:
No I think we need to include both values. So probably this instead:

```rb
      original_rubyopt = [ENV["RUBYOPT"], @envs["RUBYOPT"]].compact.join(" ")
```